tests/functional/aws-node-sdk/test/bucket/listingCornerCases.js

```diff
@@ -0,0 +1,446 @@
+const AWS = require('aws-sdk');
+const async = require('async');
+const assert = require('assert');
+
+import getConfig from '../support/config';
+
+function cutAttributes(data) {
+    const newContent = [];
+    const newPrefixes = [];
+    data.Contents.forEach(item => {
+        newContent.push(item.Key);
+    });
+    /* eslint-disable no-param-reassign */
+    data.Contents = newContent;
+    data.CommonPrefixes.forEach(item => {
+        newPrefixes.push(item.Prefix);
+    });
+    /* eslint-disable no-param-reassign */
+    data.CommonPrefixes = newPrefixes;
+    if (data.NextMarker === '') {
+        /* eslint-disable no-param-reassign */
+        delete data.NextMarker;
+    }
+    if (data.EncodingType === '') {
+        /* eslint-disable no-param-reassign */
+        delete data.EncodingType;
+    }
+    if (data.Delimiter === '') {
+        /* eslint-disable no-param-reassign */
+        delete data.Delimiter;
+    }
+}
+
+const Bucket = `bucket-listing-corner-cases-${Date.now()}`;
+
+const objects = [
+    { Bucket, Key: 'Pâtisserie=中文-español-English', Body: '' },
+    { Bucket, Key: 'notes/spring/1.txt', Body: '' },
+    { Bucket, Key: 'notes/spring/2.txt', Body: '' },
+    { Bucket, Key: 'notes/spring/march/1.txt', Body: '' },
+    { Bucket, Key: 'notes/summer/1.txt', Body: '' },
+    { Bucket, Key: 'notes/summer/2.txt', Body: '' },
+    { Bucket, Key: 'notes/summer/august/1.txt', Body: '' },
+    { Bucket, Key: 'notes/year.txt', Body: '' },
+    { Bucket, Key: 'notes/yore.rs', Body: '' },
+    { Bucket, Key: 'notes/zaphod/Beeblebrox.txt', Body: '' },
+];
+
+describe('Listing corner cases tests', () => {
+    let s3;
+    before(done => {
+        const config = getConfig('default', { signatureVersion: 'v4' });
+        s3 = new AWS.S3(config);
+        s3.createBucket(
+            { Bucket }, (err, data) => {
+                if (err) {
+                    done(err, data);
+                }
+                async.eachLimit(
```

Comment:
Super minor suggestion. This can be shortened to
```js
 async.eachLimit(objects, 999, s3.putObject, done);
```
Also, curious. Why 999 when only 10 objects are being inserted? If there is no reason, `async.each` can be used in it's place (just removes the confusion of what 999 is doing here).